package.json

```diff
@@ -195,5 +196,9 @@
   "engines": {
     "node": ">=12 <=16",
     "npm": "6"
+  },
+  "resolutions": {
+    "semver-regex": "4.0.2",
+    "lodash": "4.17.21"
```

Comment:
Lodash package version is not getting enforced for me locally, Even nuking the modules and reinstalling didnt help. 
```
      - "lodash": "4.17.21",
      + "lodash": "^4.16.6",
```

Reply:
@vigneshshanmugam did you try `npm install` or `npm ci`? Can you send me the Node version you're using?

With `npm ci` the dependencies should indeed be frozen. With `install`, on the other hand, the dependencies down the tree will still be updated, as that's NPM's default behaviour.

Having said that, it just came to mind that then running `npm-force-resolutions` as a `preinstall` doesn't really make much sense as you should really be locking dependencies for `npm ci`, not for `install`. Therefore, I've added it as just another NPM script.

I was thinking about making it a pre-commit hook, what do you think?

Reply:
I tried npm install. 

>Having said that, it just came to mind that then running npm-force-resolutions as a preinstall doesn't really make much sense as you should really be locking dependencies for npm ci, not for install

++ I think this is a better approach in general. 

> I was thinking about making it a pre-commit hook, what do you think?

Yeah that should be good.